nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -9,12 +9,12 @@ let
   efi = config.boot.loader.efi;
 
   realGrub = if cfg.version == 1 then pkgs.grub
-    else if cfg.zfsSupport then pkgs.grub2.override { zfsSupport = true; }
+    else if cfg.zfsSupport then pkgs.grub2_light.override { zfsSupport = true; }
```

Comment:
seems we should just be using regular grub in this case?

Reply:
especially since:

```
2826  grub2_light = grub2.override {
2827    zfsSupport = false;
2828  };
```

Reply:
https://github.com/NixOS/nixpkgs/blob/129f94fb38787ef866805e0f38e4b5df8050c20c/pkgs/top-level/all-packages.nix#L2818-L2828

Reply:
The default grub has ZFS support enabled

Reply:
Oh I get what you mean, overriding light with zfs = true is the same as the normal one, gotcha

Reply:
Yes, and that is what we want here:

 +    else if cfg.zfsSupport then pkgs.grub2_light.override { zfsSupport = true; }

Reply:
Ideally the `grub2_light` variant should be the one used by the NixOS module, so that only the relevant features are enabled, because permuting all the features that might be available through the module as package definitions in `all-packages.nix` would be even more overkill.

Besides (and unrelated) these `grub2.override`s should really be `self.grub2.override` so they can be overridden via overlays.